lib/spack/spack/database.py

```diff
@@ -309,7 +309,11 @@ def reindex(self, directory_layout):
                 for spec in directory_layout.all_specs():
                     # Create a spec for each known package and add it.
                     path = directory_layout.path_for_spec(spec)
-                    self._add(spec, path, directory_layout)
+                    old_info = old_data.get(spec.dag_hash())
+                    explicit = False
```

Comment:
This default is for directories that were found, but not in the database before you reindexed. I'm not too worried about what the default is.

A new command like `spack mark` or `spack keep` could be used to query and set/unset the flag for specific builds.